cl/beacon/builder/client.go

```diff
@@ -0,0 +1,173 @@
+package builder
```

Comment:
All good, some unit tests would be awesome, you can probably test by making an http test server which mocks the actual real-life behaviour

Reply:
done by overwrite roundtripper